SLMM.Domain/LawnMap/RectLawnMap.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Drawing;
+
+namespace SLMM.Domain.LawnMap
+{
+    public class RectLawnMap : IRectLawnMap
+    {
+        public Rectangle Geometry { get; set; }
+
+        public RectLawnMap(Rectangle geometry)
+        {
+            Geometry = geometry;
+
+            if (Geometry.IsEmpty)
+                throw new ArgumentOutOfRangeException(nameof(geometry));
+        }
+
+        public bool CanMove(Point coordinates)
+        {
+            return Geometry.Contains(coordinates);
+        }
+
+        public bool CanMove(in Position position, int steps)
+        {
+            EnsurePositionWithinLawn(position);
+
+            return CanMove(AddSteps(position, steps));
+        }
+
+        public bool TryMove(in Position position, int steps, out Position newPosition)

```

Comment:
As I mentioned above, I do not like the fact that responsibiliby of moving is put inside the Lawn - this should be iun the Mower.

Lawn provides information avout whether it is possible to move into given position, and that's should be it as far as the Lawn is concerned.

Reply:
See above comment regarding TryMove.